File: 00. Estructura/01. Lenguajes de proposito general/02. Programación básica/07. Funciones/05. Con paramatros y con retorno simple/MathLab/Con paramatros y con retorno simple.m

% Llamado
'Contexto' funcion(valor1, valor2, valorN) 'Contexto'

% Llamado en asignación
variable = funcion(valor1, valor2, valorN)

% Creación con parametros limitados
function retorno = funcion(var1, var2, varN)
    % Contenido
    retorno = valor
end

% Creación con parametros ilimitados
function retorno = funcion(varargin)
    % Contenido
    'Contexto' varargin{3} 'Contexto'    % Obtener parametro en la posición especificada
    % Contenido
    retorno = valor
end

% Eliminación
NO SOPORTADO

File: 01. Lenguajes de proposito general/01. Programación básica/08. Funciones/05. Con paramatros y con retorno simple/Java/Con paramatros y con retorno simple.java
package prueba;

public class Prueba {

    // Creación con parametros limitados
    public static void funcion(tipo var1, tipo var2, tipo varN) {
        // Contenido
        return valor;
    }

    // Creación con parametros ilimitados
    public static void funcion(tipo ...variables) {
        // Contenido
        return valor;
    }

    // Creación con parametros por defecto (Los valores por defecto de pueden omitir en el llamado)
    NO SOPORTADO

    public static void main(String[] args) {
        
        // Llamado
        /*Contexto*/ funcion(valor1, valor2, valorN) /*Contexto*/

        // Llamado en asignación
        tipo variable = funcion(valor1, valor2, valorN);

        // Eliminación
        NO SOPORTADO (EL GARBAGE COLLECTOR LA ELIMINA AUTOMATICAMENTE CUANDO YA ES INACCESIBLE)

    }
    
}
